packages/core/upload/admin/src/utils/getAllowedFiles.ts

```diff
@@ -1,16 +1,29 @@
-import toSingularTypes from './toSingularTypes';
-
+import { toSingularTypes } from './toSingularTypes';
+import type { File } from '../../../shared/contracts/files';
 /**
  * Returns the files that can be added to the media field
  * @param {Object[]} pluralTypes Array of string (allowedTypes)
  * @param {Object[]} files Array of files
  * @returns Object[]
  */
-const getAllowedFiles = (pluralTypes, files) => {
+
+interface AllowedFiles extends File {
+  // documentId, isSelectable, locale, type
```

Comment:
looks like the comment was forgotten. Also the interface name should be singular